src/array/c_array.cpp

```diff
@@ -41,7 +41,8 @@ CArray& CArray::operator=(CArray&& other) {
 }
 
 Array* CArray::clone() const {
-  return new CArray(aero_bridge_c_array_clone(c_ptr_));
+  if (c_ptr_) return new CArray(aero_bridge_c_array_clone(c_ptr_));
```

Comment:
Under what circumstances should an existing array have null data?

Reply:
good point - I'm too used to how it's not really possible in Fortran to prevent someone from using an uninitialized variable

Reply:
Makes sense. If we want to be careful here, we can `AERO_ASSERT(c_ptr_ != nullptr);` at the top.